com.unity.perception/Runtime/GroundTruth/PerceptionCamera.cs

```diff
@@ -424,21 +426,26 @@ void OnSimulationEnding()
 
         void OnBeginCameraRendering(ScriptableRenderContext _, Camera cam)
         {
-            if (cam != m_AttachedCamera)
-                return;
-            if (!SensorHandle.ShouldCaptureThisFrame)
-                return;
-            //there are cases when OnBeginCameraRendering is called multiple times in the same frame. Ignore the subsequent calls.
-            if (m_LastFrameCaptured == Time.frameCount)
+            if (!ShouldCallLabelers(cam, m_LastFrameCaptured))
                 return;
 
             m_LastFrameCaptured = Time.frameCount;
-#if UNITY_EDITOR
-            if (UnityEditor.EditorApplication.isPaused)
-                return;
-#endif
             CaptureRgbData(cam);
+            CallOnLabelers(l => l.InternalOnBeginRendering());
+        }
+
+        void OnEndCameraRendering(ScriptableRenderContext _, Camera cam)
+        {
+            if (!ShouldCallLabelers(cam, m_LastFrameEndRendering))
+                return;
 
+            m_LastFrameEndRendering = Time.frameCount;
+            CallOnLabelers(l => l.InternalOnEndRendering());
+        }
+
+
+        private void CallOnLabelers(Action<CameraLabeler> action)
```

Comment:
I'm not sure about this name, how about something like: NotifyLabelers(Action<CameraLabeler> action)